src/components/nfts/NftCollections/index.tsx

```diff
@@ -1,61 +1,109 @@
-import { useState, useEffect } from 'react'
+import { useEffect, useMemo, useState } from 'react'
 import { type SafeCollectibleResponse } from '@safe-global/safe-gateway-typescript-sdk'
-import { Box, CircularProgress } from '@mui/material'
-import useCollectibles from '@/hooks/useCollectibles'
+import { Box, Button, CircularProgress, SvgIcon, Typography } from '@mui/material'
 import ErrorMessage from '@/components/tx/ErrorMessage'
-import InfiniteScroll from '@/components/common/InfiniteScroll'
 import PagePlaceholder from '@/components/common/PagePlaceholder'
 import NftIcon from '@/public/images/common/nft.svg'
-import NftTransferModal from '@/components/tx/modals/NftTransferModal'
+import ArrowIcon from '@/public/images/common/arrow-nw.svg'
+import NftBatchModal from '@/components/tx/modals/NftBatchModal'
 import NftGrid from '../NftGrid'
 import useIsGranted from '@/hooks/useIsGranted'
+import useCollectibles from '@/hooks/useCollectibles'
+import InfiniteScroll from '@/components/common/InfiniteScroll'
 
 const NftCollections = () => {
+  const [pageUrl, setPageUrl] = useState<string>()
+  const [nftPage, error, loading] = useCollectibles(pageUrl)
   const [allNfts, setAllNfts] = useState<SafeCollectibleResponse[]>([])
-  const [currentPageUrl, setCurrentPageUrl] = useState<string | undefined>()
-  const [sendNft, setSendNft] = useState<SafeCollectibleResponse>()
+  const [selectedNfts, setSelectedNfts] = useState<SafeCollectibleResponse[]>([])
+  const [sendNfts, setSendNfts] = useState<SafeCollectibleResponse[]>()
+  const [filter, setFilter] = useState<string>('')
   const isGranted = useIsGranted()
 
-  const [page, error, loading] = useCollectibles(currentPageUrl)
+  const onSelect = (token: SafeCollectibleResponse) => {
+    setSelectedNfts((prev) => (prev.includes(token) ? prev.filter((t) => t !== token) : prev.concat(token)))
+  }
+
+  const filteredNfts = useMemo(() => {
+    return allNfts.filter((nft) => nft.tokenName.toLowerCase().includes(filter))
+  }, [allNfts, filter])
 
-  /* Aggregate NFTs from all pages */
   useEffect(() => {
-    if (page) {
-      setAllNfts((prev) => prev.concat(page.results))
+    if (nftPage) {
+      setAllNfts((prev) => prev.concat(nftPage.results))
     }
-  }, [page])
+  }, [nftPage])
 
   /* No NFTs to display */
-  if (!allNfts.length && !loading && !error) {
+  if (nftPage && !nftPage.results.length) {
     return <PagePlaceholder img={<NftIcon />} text="No NFTs available or none detected" />
   }
 
   return (
     <>
-      {/* Aggregated NFTs grouped by collection */}
-      {allNfts.length > 0 && (
-        <NftGrid collectibles={allNfts} onSendClick={isGranted ? (nft) => setSendNft(nft) : undefined} />
+      {allNfts?.length > 0 && (
+        <>
+          {/* Mass send button */}
+          <Box mb={2} display="flex" alignItems="center" gap={1}>
```

Comment:
Should we make this sticky?
Especially useful with InfiniteScrolling IMO.

Reply:
Made it sticky, thanks for the idea!